Resources/public/js/views/services/plugins/ez-userdraftsplugin.js

```diff
@@ -0,0 +1,204 @@
+         *
+         * @method _collectDraftsData
+         * @protected
+         * @param versions {Array} list of versions
+         * @return {Y.Promise}
+         */
+        _loadDraftContentInfo: function (versions) {
+            var drafts = this.get('drafts'),
+                promises = [];
+
+            versions.forEach(Y.bind(function (version) {
+                drafts.push({
+                    version: version,
+                    contentType: null,
+                    contentInfo: null
+                });
+
+                promises.push(new Y.Promise(Y.bind(function (resolve, reject) {
+                    var contentInfo = new Y.eZ.ContentInfo({id: version.get('resources.Content')});
+
+                    contentInfo.load({api: this.get('host').get('capi')}, function (error) {
+                        if (error) {
+                            reject(error);
+
+                            return;
+                        }
+
+                        resolve(contentInfo);
+                    });
+                }, this)));
+            }, this));
+
+            return Y.Promise.all(promises);
+        },
+
+        /**
+         * Loads content types for each content version
+         *
+         * @method _loadDraftContentType
+         * @protected
+         * @param data {Array} list of content
+         * @return {Y.Promise}
+         */
+        _loadDraftContentType: function (data) {
+            var drafts = this.get('drafts'),
+                promises = [];
+
+            data.forEach(Y.bind(function (contentInfo, index) {
+                var contentType = new Y.eZ.ContentType({id: contentInfo.get('resources.ContentType')});
+
+                drafts[index].contentInfo = contentInfo;
+
+                promises.push(new Y.Promise(Y.bind(function (resolve, reject) {
+                    contentType.load({api: this.get('host').get('capi')}, function (error) {
+                        if (error) {
+                            reject(error);
+
+                            return;
+                        }
+
+                        resolve(contentType);
+                    });
+                }, this)));
+            }, this));
+
+            return Y.Promise.all(promises);
+        },
+
+        /**
+         * Creates a content type model, fills with data and assigns it to each draft
+         *
+         * @method _fillDraftContentTypeModel
+         * @protected
+         * @param target {Y.View} event target view
+         * @param attributeName {String} name of an attribute to be updated with data
+         * @param data {Array} list of content types
+         */
+        _fillDraftContentTypeModel: function (target, attributeName, data) {
+            var drafts = this.get('drafts');
+
+            data.forEach(function (contentType, index) {
+                drafts[index].contentType = contentType;
+            });
+
+            drafts.sort(function (a, b) {
```

Comment:
since the VersionInfo array is sorted in _collectDraftsData, I'm pretty sure it's not needed to sort again here.